lib/process-services-cloud/src/lib/services/variable-mapper.sevice.spec.ts

```diff
@@ -49,15 +46,71 @@ describe('VariableMapperService', () => {
             }
         });
 
-        const viewModel = service.mapVariablesByColumnTitle([objectWithVariables], [column]);
+        objectWithVariables = {
+            variables: [variable]
+        };
+    });
 
+    it('should map variables by column title', () => {
         const expectedObjectWithVariableMap = {
             ...objectWithVariables,
             variablesMap: {
                 [column.title]: variable
             }
         };
 
+        const viewModel = service.mapVariablesByColumnTitle([objectWithVariables], [column]);
+
         expect(viewModel).toEqual([expectedObjectWithVariableMap]);
     });
+
+    describe('should map correct column type according to process variable type in case of', () => {
+        it('date type', () => {
+            variable.type = 'boolean';
+
+            const viewModel = service.mapVariablesByColumnTitle([objectWithVariables], [column]);
+
+            expect(viewModel[0].variablesMap[column.title].type).toEqual('boolean');
+        });
+
+        it('integer type', () => {
+            variable.type = 'integer';
+
+            const viewModel = service.mapVariablesByColumnTitle([objectWithVariables], [column]);
+
+            expect(viewModel[0].variablesMap[column.title].type).toEqual('text');
+        });
+
+        it('string type', () => {
+            variable.type = 'string';
+
+            const viewModel = service.mapVariablesByColumnTitle([objectWithVariables], [column]);
+
+            expect(viewModel[0].variablesMap[column.title].type).toEqual('text');
+        });
+
+        it('date type', () => {
+            variable.type = 'date';
+
+            const viewModel = service.mapVariablesByColumnTitle([objectWithVariables], [column]);
+
+            expect(viewModel[0].variablesMap[column.title].type).toEqual('date');
+        });
+
+        it('datetime type', () => {
+            variable.type = 'datetime';
+
+            const viewModel = service.mapVariablesByColumnTitle([objectWithVariables], [column]);
+
+            expect(viewModel[0].variablesMap[column.title].type).toEqual('date');
+        });
+
+        it('other types', () => {
+            variable.type = 'custom';
+
+            const viewModel = service.mapVariablesByColumnTitle([objectWithVariables], [column]);
+
+            expect(viewModel[0].variablesMap[column.title].type).toEqual('text');
```

Comment:
please create function to avoid dupliaction also here, similar we you did for cell tests